lldb/source/Plugins/Language/Swift/SwiftFormatters.cpp

```diff
@@ -1320,6 +1320,178 @@ class TaskGroupSyntheticFrontEnd : public SyntheticChildrenFrontEnd {
   // Cache and storage of constructed child values.
   std::vector<ValueObjectSP> m_children;
 };
+
+class ActorSyntheticFrontEnd : public SyntheticChildrenFrontEnd {
+public:
+  ActorSyntheticFrontEnd(lldb::ValueObjectSP valobj_sp)
+      : SyntheticChildrenFrontEnd(*valobj_sp.get()) {
+    bool is_64bit = false;
+    if (auto target_sp = m_backend.GetTargetSP())
+      is_64bit = target_sp->GetArchitecture().GetTriple().isArch64Bit();
+
+    std::optional<uint32_t> concurrency_version;
+    if (auto process_sp = m_backend.GetProcessSP())
+      concurrency_version =
+          SwiftLanguageRuntime::FindConcurrencyDebugVersion(*process_sp);
+
+    m_is_supported_target = is_64bit && concurrency_version.value_or(0) == 1;
+    if (!m_is_supported_target)
+      return;
+
+    auto target_sp = m_backend.GetTargetSP();
+    auto ts_or_err =
+        target_sp->GetScratchTypeSystemForLanguage(eLanguageTypeSwift);
+    if (auto err = ts_or_err.takeError()) {
+      LLDB_LOG_ERROR(GetLog(LLDBLog::DataFormatters | LLDBLog::Types),
+                     std::move(err),
+                     "could not get Swift type system for Task synthetic "
+                     "provider: {0}");
+      return;
+    }
+    m_ts = llvm::dyn_cast_or_null<TypeSystemSwiftTypeRef>(ts_or_err->get());
+  }
+
+  llvm::Expected<uint32_t> CalculateNumChildren() override {
+    return m_is_supported_target ? 1 : 0;
+  }
+
+  lldb::ValueObjectSP GetChildAtIndex(uint32_t idx) override {
+    if (!m_is_supported_target || idx != 0)
+      return {};
+
+    if (!m_unprioritised_jobs_sp) {
+      std::string mangled_typename =
+          mangledTypenameForTasksTuple(m_job_addrs.size());
+      CompilerType tasks_tuple_type =
+          m_ts->GetTypeFromMangledTypename(ConstString(mangled_typename));
+      DataExtractor data{m_job_addrs.data(),
+                         m_job_addrs.size() * sizeof(addr_t),
+                         endian::InlHostByteOrder(), sizeof(void *)};
+      m_unprioritised_jobs_sp = ValueObject::CreateValueObjectFromData(
+          "unprioritised_jobs", data, m_backend.GetExecutionContextRef(),
+          tasks_tuple_type);
+    }
+    return m_unprioritised_jobs_sp;
+  }
+
+  size_t GetIndexOfChildWithName(ConstString name) override {
+    if (m_is_supported_target && name == "unprioritised_jobs")
+      return 0;
+    return UINT32_MAX;
+  }
+
+  lldb::ChildCacheState Update() override {
+    if (!m_is_supported_target)
+      return ::eReuse;
+
+    m_job_addrs.clear();
+
+    if (!m_task_type)
+      if (auto target_sp = m_backend.GetTargetSP()) {
+        if (auto ts_or_err = target_sp->GetScratchTypeSystemForLanguage(
+                eLanguageTypeSwift)) {
+          if (auto *ts = llvm::dyn_cast_or_null<TypeSystemSwiftTypeRef>(
+                  ts_or_err->get()))
+            // TypeMangling for "Swift.UnsafeCurrentTask"
```

Comment:
FWIW, the nice thing to do here would be to wrap this in a TypeSystemSwift::GetUnsafeCurrentTaskType() helper

Reply:
agreed, should I do that as part of this PR? I plan to do some refactoring like this after the fact, but I can just as easily perform some refactorings in this PR.

Reply:
Both options are fine.